File: Frontend/VIAProMa/Assets/Scripts/DataModel/RequirementsBazaar/Category.cs
using i5.VIAProMa.UI.ListView.Core;
using System;

namespace i5.VIAProMa.DataModel.ReqBaz
{
    /// <summary>
    /// Represents a category in the Requirements Bazaar
    /// </summary>
    [Serializable]
    public class Category : IListViewItemData
    {

        /// <summary>
        /// The values for a category
        /// </summary>
        public int id;
        public string name;
        public string description;
        public int projectId;
        public ReqBazUser leader;
        public string creationDate;
        public string lastUpdatedDate;
        public int numberOfRequirements;
        public int numberOfFollowers;

        /// <summary>
        /// Creates a blank category
        /// </summary>
        public Category()
        {
        }

        /// <summary>
        /// The Instantiator of a category
        /// </summary>
        /// <param name="id">The numeric id of the category</param>
        /// <param name="name">The name of the category</param>
        public Category(int id, string name)
        {
            this.id = id;
            this.name = name;
        }
    }
}

File: Backend/immersive_project_management_service/src/main/java/i5/las2peer/services/immersiveProjectManagementService/i5/las2peer/services/immersiveProjectManagementService/dataModel/requirementsBazaar/Category.java
package i5.las2peer.services.immersiveProjectManagementService.i5.las2peer.services.immersiveProjectManagementService.dataModel.requirementsBazaar;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Created by bened on 09.05.2019.
 */
//@JsonFilter("shortFilter")
@JsonIgnoreProperties(ignoreUnknown = true)
public class Category {
    public int id;
    public String name;
    public String description;
    public int projectId;
    public ReqBazUser leader;
    public String creationDate;
    public String lastUpdatedDate;
    public int numberOfRequirements;
    public int numberOfFollowers;
}
